source/math/FloatMatrix.ooc

```diff
@@ -236,6 +236,48 @@ FloatMatrix: cover {
 		this free(Owner Receiver)
 		result
 	}
+	// Thomas algorithm for tridiagonal systems
+	// https://en.wikipedia.org/wiki/Tridiagonal_matrix_algorithm
+	solveTridiagonal: func (y: This) -> This {
+		diagonal := this take() getDiagonal() take()
+		upperDiagonal := this take() getDiagonal(1, 0) take()
+		lowerDiagonal := this take() getDiagonal(0, 1) take()
+		rightHandSide := y take() copy() take()
+		d := diagonal elements
+		u := upperDiagonal elements
+		l := lowerDiagonal elements
```

Comment:
It would be more OK to have `i` instead of `index` if it still gets too bloated, though.